draft-lenders-core-coap-dtls-svcb.md

```diff
@@ -102,7 +102,103 @@ same manner as for any other service secured with transport layer security, as
+
+## Request construction
+
+The steps through which the CoAP request is constructed are as follows:
+
+* Select a CoAP transport based on per-record information, falling back to a default for the named service if one is defined.
+
+  In particular, select CoAP over UDP if the ALPN is "co", and CoAP over TCP if the ALPN is "coap".
+  Note that the use of ALPNs in the presence of a default transport for the service is also influenced by the no-default-alpn parameter as per {{Section 7.1 of RFC9460}}.
+
+  If multiple transports are available in a single record,
+  transports tried in sequence, following the rules of {{Section 7.1.2 of RFC9460}}.
+
+* Send the request to the address indicated in additional information about the target,
+  e.g. from a AAAA record associated with the target or an ipv6hint in the service parameters,
+  falling back to an attempt to resolve the names.
+
+* If a port is indicated in the service parameters, use that as the destination port;
+  otherwise, the destination port is the default port of the transport.
+
+* If the queried service name indicates a particular scheme, set that scheme in the Proxy-Scheme option.
+
+* Set the queried host name as the Uri-Host option.
+  The option should be elided if that value is the default value on that transport.
+
+* If the queried port differs from either the destination port or the used protocol's default port,
+  set the Uri-Port option to the queried port.
+
+  \[ There may need to be a rule that sets the queried port to the default port of the scheme implied by the service name if there is one;
+  that is best handle when better understanding other use cases. \]
+
+* If the queried service specifies an option through which a particular resource can be set,
+  add a Uri-Path option for each value in that list.
+  It is recommended that such options are serialized as a CBOR sequence of text strings.
+
+Note that when services names that do not indicate a particular scheme (such as "dns")
+are used with multiple ALPNs,
+create implicit URI aliasing between the two URIs those requests can be composed into.
+
+## Example
+
+A host receives an IPv6 Route Advertisement (RA) for Encrypted DNS as described in {{Section 6 of -dnr}}.
+The RA contains the priority 1, authenticated-domain-name "dns1.example.com", IPv6 address 2001:db8:53::1 and service parameters alpn="coap",port="61616",keycpa9=\143dns\146server (where keycpa9 may also written as docpath="dns","server" depending on whether that option defines a zone file format).
+
+The Encrypted DNS RA is implied to be a response to an SVCB request to "\_dns.dns1.example.com".
+The "dns" service has no implied default transport for CoAP, and the "docpath" is the service parameter that provides path construction for the request path.
+
+A DNS request is therefore sent by establishing a CoAP over TLS connection
+with 2001:db8:53::1 on port 61616.
+The name "dns1.example.com" is sent through the SNI extension.
```

Comment:
Not sure this distracts lesser informed reader somehow, since this is an example and “SNI extension” was not previously mentioned.